### ISSUE TITLE
ERROR @wdio/local-runner: Failed launching test session: Error: unknown error

### ISSUE THREAD

**Issue by Amrkamel1** [webdriverio/webdriverio#6214]
Console log error is showing after the test is done when using WinAppDriver services 

**Environment (please complete the following information):**
```
 "keywords": [],
  "author": "aMr kaMel",
  "license": "ISC",
  "dependencies": {
    "@wdio/cli": "^6.4.7",
    "appium": "^1.18.3",
    "eslint": "^7.8.1",
    "frisby": "^2.1.3",
    "webdriverio": "^6.4.7"
  },
  "devDependencies": {
    "@wdio/selenium-standalone-service": "^6.4.7",
    "@wdio/appium-service": "^6.4.7",
    "wdio-chromedriver-service": "^6.0.4",
    "wdio-winappdriver-service": "^0.1.0",
    "@wdio/concise-reporter": "^6.4.7",
    "@wdio/dot-reporter": "^6.4.7",
    "@wdio/local-runner": "^6.4.7",
    "@wdio/mocha-framework": "^6.4.7",
    "@wdio/allure-reporter": "^6.4.7",
    "@wdio/spec-reporter": "^6.4.7",
    "@wdio/sync": "^6.4.7",
    "chai": "^4.2.0",
    "mocha": "^8.1.3"
  }
```

**Config of WebdriverIO**
```
exports.config = {

   specs: [
    'src/test/specs/03_WinApp_Suite.js'
  ],

  capabilities: [{
    port: 4729,
    path: '/wd/hub',
    app: 'C:/Users/amrka/AppData/Local/Programs/Evernote/Evernote.exe',
    platformName: 'Windows',
    deviceName: 'WS-test'
  }],

  services: [
    ['winappdriver', {
        args: ['localhost', '4729/wd/hub']
    }]
  ],

  maxInstances: 2,
  waitforTimeout: 13000,
  connectionRetryTimeout: 90000,
  framework: 'mocha',
  mochaOpts: {
    ui: 'bdd',
    timeout: 120000,
    retries: 0
  },

    afterTest: function(test, context, { error, result, duration, passed, retries }) {
        if (error !== undefined) {
            browser.takeScreenshot();
          }
    },

    reporters: ['dot','spec',['allure', {
        outputDir: 'allure-results',
        disableWebdriverStepsReporting: true,
        disableWebdriverScreenshotsReporting: false,
        disableMochaHooks: true
    }]],
}
```

**Describe the bug**
Unknown error no matter if it block is passing or failing is show upon test completion leading fail for test spec 

**To Reproduce**
Using common config as shown above for winAppdriver services

[Include code or an example repository that can easily be set up]

**Expected behavior**
Error should be handled 

**Log**

```
[0-0] 2020-12-10T16:09:14.853Z INFO webdriver: COMMAND deleteSession()
[0-0] 2020-12-10T16:09:14.854Z INFO webdriver: [DELETE] http://localhost:4729/wd/hub/session/59F0BBF8-F7A9-432B-A1B8-66B13EC4A248
[2020-12-10T16:09:14.870Z]  INFO: electron.main/13928 on WS-AMRKA(C:\Users\amrka\AppData\Local\Programs\Evernote\resources\app.asar\node_modules\conduit-utils\node_modules\en-ts-utils\src\logger.ts:222 in info):  onAppBeforeQuit
[2020-12-10T16:09:14.875Z]  INFO: electron.main/13928 on WS-AMRKA(C:\Users\amrka\AppData\Local\Programs\Evernote\resources\app.asar\node_modules\conduit-utils\node_modules\en-ts-utils\src\logger.ts:222 in info):  clearWindow for id: 2
[2020-12-10T16:09:14.877Z]  INFO: electron.main/13928 on WS-AMRKA(C:\Users\amrka\AppData\Local\Programs\Evernote\resources\app.asar\node_modules\conduit-utils\node_modules\en-ts-utils\src\logger.ts:222 in info):  Cleaning up from 'unload' event
[0-0] 2020-12-10T16:09:14.921Z DEBUG webdriver: request failed due to missing body
[0-0] 2020-12-10T16:09:14.921Z ERROR webdriver: Request failed with status 200 due to Error: unknown error
[0-0] 2020-12-10T16:09:14.923Z ERROR @wdio/local-runner: Failed launching test session: Error: unknown error
    at Object.getErrorFromResponseBody (C:\Users\amrka\Desktop\Contributions\Ultimate-Framework\node_modules\webdriver\build\utils.js:92:16)
    at WebDriverRequest._request (C:\Users\amrka\Desktop\Contributions\Ultimate-Framework\node_modules\webdriver\build\request.js:133:31)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async Browser.runCommandWithHooks (C:\Users\amrka\Desktop\Contributions\Ultimate-Framework\node_modules\@wdio\sync\build\wrapCommand.js:70:25)
    at Browser.runCommandWithHooks (C:\Users\amrka\Desktop\Contributions\Ultimate-Framework\node_modules\@wdio\sync\build\wrapCommand.js:65:24)
    at Browser.wrapCommandFn (C:\Users\amrka\Desktop\Contributions\Ultimate-Framework\node_modules\@wdio\sync\build\wrapCommand.js:44:44)
    at Runner.endSession (C:\Users\amrka\Desktop\Contributions\Ultimate-Framework\node_modules\@wdio\runner\build\index.js:190:30)
    at Runner.run (C:\Users\amrka\Desktop\Contributions\Ultimate-Framework\node_modules\@wdio\runner\build\index.js:98:24)
[2020-12-10T16:09:14.934Z]  INFO: electron.main/13928 on WS-AMRKA(C:\Users\amrka\AppData\Local\Programs\Evernote\resources\app.asar\node_modules\conduit-utils\node_modules\en-ts-utils\src\logger.ts:222 in info):  ----- App has been closed ----- {"reason":"exit","code":0}
2020-12-10T16:09:14.949Z DEBUG @wdio/local-runner: Runner 0-0 finished with exit code 1
[0-0] FAILED in C:/Users/amrka/AppData/Local/Programs/Evernote/Evernote.exe - C:\Users\amrka\Desktop\Contributions\Ultimate-Framework\src\test\specs\03_WinApp_Suite.js
2020-12-10T16:09:14.952Z INFO @wdio/cli:launcher: Run onComplete hook
2020-12-10T16:09:14.953Z DEBUG winappdriver-service: WinAppDriver (pid: 7760) is killed
```

**Comment by mgrybyk** [comment]
Let me clarify a bit. The case above is one of the issues happening because of fallback to devtools protocol mechanish.

@christian-bromann do you think it worst disabling fallback to devtools protocol or check capabilities/config before trying doing that?
examples when  it doesn't make sense to switch to devtools protocol:
- if selenium standalone service is used and devtools protocol is not specified explicitly
- if there is no browserName in capabilities

**Comment by christian-bromann** [comment]
> * if selenium standalone/chromedriver service is used and devtools protocol is not specified explicitly

This should not be a problem as these services update the config to have capabilities properly connect to the driver and therefor make WebdriverIO automatically pick the WebDriver protocol.

> * if there is no browserName in capabilities (current issue)

Yeah, this could be problematic. We should also default to WebDriver if we have an indication of a mobile remote environment.

**Comment by christian-bromann** [comment]
fixed in #6929
